### PR TITLE
Generate sequence

### PR DESCRIPTION
**Pull request by PawelYak**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: generate-sequence
🏅 Granted 5 XP for @PawelYak

**Comment by ziobron** [comment]
⏱ 3 XP extra for @PawelYak for on-time delivery

File: homework/generate-sequence/generateSequence.hpp [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
 #include <vector>
 
 std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
-}
+  std::vector<int> vec{step};
+
+  if (count <= 0) {
+    return std::vector<int>{};
```
Comment (nauka-programowania-MA) [inline homework/generate-sequence/generateSequence.hpp:9]: just return {}

File: homework/generate-sequence/generateSequence.hpp [hunk 2]
```diff
@@ -1,7 +1,19 @@
 #pragma once
+#include <math.h>
 #include <vector>
 
 std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
-}
+  std::vector<int> vec{step};
```
Comment (nauka-programowania-MA) [inline homework/generate-sequence/generateSequence.hpp:6]: what if step is negate? You will get execption here

File: homework/generate-sequence/generateSequence.hpp [hunk 3]
```diff
@@ -1,7 +1,19 @@
...
+
+  if (count <= 0) {
+    return std::vector<int>{};
+  } else if (step == 0)
```
Comment (nauka-programowania-MA) [inline homework/generate-sequence/generateSequence.hpp:10]: If you return variable here, you don't need to use `else if` or `else`

File: homework/generate-sequence/generateSequence.hpp [hunk 4]
```diff
@@ -1,7 +1,19 @@
...
+    for (auto i = 3; i < count; i++) {
+      vec.push_back(step);
+    }
+
+  for (auto i = 1; i < count; i++) {
+    vec.push_back(step + vec[i - 1]);
+  }
```
Comment (nauka-programowania-MA) [inline homework/generate-sequence/generateSequence.hpp:17]: Avoid 2 loop. You can do it in one.

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**